NX-684: Registry root expiration fixes

cstamas opened the pull request:
Multiple problems fixes and improvements added:
- NX Core handled NPM MD as expired (as it sees it as coming from localstorage/cache)
- NPM Proxy MD service had a bug, that considered NPM reg root always as expired as it was incomplete
- improved logging with some TRACE logs
- added IT ensuring that once cached, registry root is not re-fetched

Issue
https://issues.sonatype.org/browse/NX-684

cstamas committed:
NX-684: Registry root expiration fixes

Multiple problems fixes and improvements added:
* NX Core handled NPM MD as expired (as it sees it as coming from localstorage/cache)
* NPM Proxy MD service had a bug, that considered NPM reg root always as expired as it was incomplete
* improved logging with some TRACE logs
* added IT ensuring that once cached, registry root is not re-fetched

scarlucci commented on nexus-npm-repository-plugin/src/main/java/com/bolyuba/nexus/plugin/npm/metadata/internal/ProxyMetadataServiceImpl.java:
```diff
@@ -86,7 +86,12 @@ protected PackageRootIterator doGenerateRegistryRoot(final PackageRequest reques
       final long now = System.currentTimeMillis();
       if (registryRoot == null || isExpired(registryRoot, now)) {
```
Do we need a lock here to prevent a second request from attempting to download the registry root while we are already downloading it?

cstamas replied:
True, fixing it. The lock existing in proxy code doing the call into proxy MD service is shared, and hence, concurrently entering threads are possible.

cstamas committed:
NX-684: Assure that registry root update does not happens concurrently

cstamas committed:
NX-684: fixing path recorder lifecycle

As it was created when mock registry was. Now it's re-created
per each start, to avoid possible interleaves of recorded data
from previous runs.
